hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaRolloutHandler.java

```diff
@@ -92,19 +94,29 @@ private static String createRolloutLockKey(final String tenant) {
         return tenant + "-rollout";
     }
 
+    // run in a tenant context, i.e. contextAware.getCurrentTenant() returns the tenant
+    // the rollout is made for
     private void handleRolloutInNewTransaction(final long rolloutId, final String handlerId) {
         DeploymentHelper.runInNewTransaction(txManager, handlerId + "-" + rolloutId, status -> {
             rolloutManagement.get(rolloutId).ifPresentOrElse(
-                    rollout -> runInUserContext(rollout, () -> rolloutExecutor.execute(rollout)),
+                    rollout -> {
+                        rollout.getAccessControlContext().ifPresentOrElse(
+                            context -> // has stored context - executes it with it
+                                contextAware.runInContext(
+                                    context,
+                                    () -> rolloutExecutor.execute(rollout)),
+                            () -> // has no stored context - executes it in the tenant & user scope
```

Comment:
At this place I remembered a consequence of restoring the context again.

Let's assume the users permissions changed in the meanwhile, we will anyway set the permission present during rollout creation. I think this behaviour is fine, however it needs to be documented.

Reply:
This depends on the implementation of the SecurityContextSerializater that is used.
Implementation could, let say, does not store the user permissions (as Java serializer) but just the user id, and then to get entity base permissions from a dynamic database - so the permissions of stored context could change over time.
However, they could also be static (as java serializer). So, I think no explicit need to document.
Whoever implements  access control shall chose its serializer and have in account its features